### PR TITLE
Create Heatmap layer and add helper functions

### PR DESCRIPTION
**Pull request by LindseyWeiskopf**
Summary: I created a heatmap layer so that you can see where there are clumps of dropoff locations and absences of dropoff locations. I also created helper functions for the previous code that I had.
[lweiskopf-step-2020.appspot.com](url)***

In this PR, I created a heatmap layer which is a visualization tool for the maps API. To do this, I created an array list of LatLng objects (representing the locations) and created a heatmap instance based off of this data with other properties like radius and opacity. 

This visualization allows the user to see trends in dropoff locations throughout NYC. Though there doesn't seem to be a ton of trends in this set of data, I think that this tool would be really helpful in identifying things like food deserts or access to healthcare. I also think that it would work better on a larger geographic scale because the minimum radius seems to be a bit large.

***You might have to refresh your cache for the heatmap to show up

Thanks!

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-// Creates a map and adds it to the page with markers on compost-dropoff locations.
+// Global variables that apply across functions
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:55]: This comment, as is, is redundant. Perhaps describing why they're useful instead? Or feel free to leave them out.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
...
+var map, heatmap;
+var heatmapData = [];
+
+// Creates a map and heatmap layer
 function initMap() {
   // Coordinates centers map  around NYC
   var myLatlng = {lat: 40.763232, lng: -73.951047};
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:62]: Please make these literals constants.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
...
   var mapOptions = {
     center: myLatlng, 
-    zoom: 12,
+    zoom: 14,
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:65]: Please make this literal a constant, with a comment describing its purpose.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
...
     mapTypeControl: false 
   };
-  const map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  map = new google.maps.Map(document.getElementById('map'), mapOptions);
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:68]: Is this variable used anywhere?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
...
+  getCoords();
+  createHeatmap();
+  
+}
+
+// Iterates through coordinates and delivers to other functions
+function getCoords() {
   var image = 'http://maps.google.com/mapfiles/kml/pal4/icon47.png';
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:76]: Please assign this path to a constant for this variable to use.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
...
+}
+
+function createHeatmap() {
+  heatmap = new google.maps.visualization.HeatmapLayer({
+    data: heatmapData,
+    dissipating: false,
+    map: map,
+    radius: 1,
+    opacity: 0.5,
+    // green gradient
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:108]: Very useful comment. Thank you!

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js]: I decided to just delete the comment.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js]: Got it.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js:70]: Yes - whenever anything is added, it adds to the map object, and the heatmap is based upon this 'map' as well.

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/webapp/script.js]: I saved the image locally so now there should be a const path to the images folder

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (shaquille-hall) [reply portfolio/src/main/webapp/script.js:70]: Ahh this is global. Totally forgot. I see now. Thank you.

**Pull request merged**